websites/K/KOLEO/presence.ts

```diff
@@ -0,0 +1,980 @@
+				presenceData.smallImageKey = Assets.Buy;
+			}
+		} else if (currentPath.startsWith("/moje-konto")) {
+			if (document.querySelector(".woocommerce-form-login__submit")) {
+				presenceData.details = "Loguje się do swojego konta KOLEO.";
+				presenceData.smallImageText = "Loguje się...";
+				presenceData.smallImageKey = Assets.Writing;
+			} else {
+				presenceData.details = "Przegląda swoje konto KOLEO.";
+				presenceData.smallImageText = "Przegląda konto...";
+			}
+
+			if (currentPath.startsWith("/moje-konto/lost-password")) {
+				presenceData.details = "Resetuje swoje hasło do konta KOLEO.";
+				presenceData.smallImageText = "Resetuje hasło...";
+				presenceData.smallImageKey = Assets.Writing;
+			}
+		} else if (currentPath.startsWith("/regulamin")) {
+			presenceData.details = "Czyta regulamin sklepu KOLEO.";
+			presenceData.smallImageText = "Czyta regulamin...";
+			presenceData.smallImageKey = Assets.Reading;
+		} else if (currentPath.startsWith("/polityka-prywatnosci")) {
+			presenceData.details = "Czyta politykę prywatności sklepu KOLEO.";
+			presenceData.smallImageText = "Czyta politykę...";
+			presenceData.smallImageKey = Assets.Reading;
+		} else if (currentPath.startsWith("/francuska-11a")) {
+			presenceData.details = "Przegląda informacje kontaktowe KOLEO.";
+			presenceData.smallImageText = "Przegląda informacje...";
+		} else if (currentPath.startsWith("/sklep-stacjonarny")) {
+			presenceData.details =
+				"Przegląda informacje o sklepie stacjonarnym KOLEO.";
+			presenceData.smallImageText = "Przegląda informacje...";
+		} else if (currentPath.startsWith("/dostawa")) {
+			presenceData.details = "Przegląda informacje o dostawie w sklepie KOLEO.";
+			presenceData.smallImageText = "Przegląda informacje...";
+		} else if (
+			currentPath.startsWith("/interrail-najczesciej-zadawane-pytania")
+		) {
+			presenceData.details =
+				"Przegląda najczęściej zadawane pytania o bilety Interrail.";
+			presenceData.smallImageText = "Przegląda pytania...";
+		} else {
+			const productTitle = document.querySelector(".product_title");
+			if (productTitle) {
+				presenceData.details = `Przegląda produkt${
+					privacySetting ? "" : ` - ${productTitle.textContent}`
+				}.`;
+				presenceData.state = `${
+					privacySetting
+						? ""
+						: `Marka: ${
+								document.querySelector(".pwb-single-product-brands a")
+									.textContent
+						  } | `
+				}Cena: ${
+					document.querySelector(".price span.woocommerce-Price-amount bdi")
+						?.textContent
+				}`;
+				presenceData.smallImageText = "Przegląda produkt...";
+				if (!privacySetting)
+					presenceData.buttons = [{ label: "Zobacz produkt", url: href }];
+				else {
+					presenceData.buttons = [
+						{ label: "Zobacz katalog", url: "https://sklep.koleo.pl/katalog" },
+					];
+				}
+			} else NoPage(presenceData);
+		}
+	} else NoPage(presenceData);
+
+	presence.setActivity(presenceData);
```

Comment:
Suggestion 3:
Here we can check if the privacy setting is on & if there's a state, if so delete the state.
```suggestion
       if (privacySetting && presenceDate.state) delete presenceData.state
	presence.setActivity(presenceData);
```

Reply:
Won't be doing this because in some places i am using the state part for not privacy realted things aswell